src/app/Traits/GetExistingUser.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace NetworkRailBusinessSystems\UserLogin\Traits;
+
+use LdapRecord\Models\ActiveDirectory\User as LdapUser;
+
+trait GetExistingUser
+{
+    /**
+     * @codeCoverageIgnore
+     */
+    public static function mail($username): ?string
+    {
+        return LdapUser::query()
+            ->where('samaccountname', '=', $username)
+            ->first()?->getAttributeValue('mail')[0];
```

Comment:
Make `mail` configurable, in line with what I said regarding making this more abstract.